From fab68197d1faae2a3415e6a1ce66dd5b6fa99f75 Mon Sep 17 00:00:00 2001
From: Alexey Kodanev <aleksei.kodanev@bell-sw.com>
Date: Fri, 29 Nov 2024 14:11:01 +0000
Subject: [PATCH] ldd: process only one program interpreter segment type

It shouldn't appear more than once, and the kernel will only
process the first one anyway.
---
 ldso/dynlink.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/ldso/dynlink.c b/ldso/dynlink.c
index 6ca0cc80..e61714e6 100644
--- a/ldso/dynlink.c
+++ b/ldso/dynlink.c
@@ -1953,8 +1953,10 @@ void __dls3(size_t *sp, size_t *auxv)
 		 * linker had ldd not taken its place. */
 		if (ldd_mode) {
 			for (i=0; i<app.phnum; i++) {
-				if (app.phdr[i].p_type == PT_INTERP)
+				if (app.phdr[i].p_type == PT_INTERP) {
 					ldso.name = laddr(&app, app.phdr[i].p_vaddr);
+					break;
+				}
 			}
 			int needed_so = 0;
 			for (i=0; app.dynv[i]; i+=2) {
-- 
2.25.1

